internal/core/items/text.rs

```diff
@@ -454,10 +454,12 @@ impl Item for TextInput {
             FocusEvent::FocusIn | FocusEvent::WindowReceivedFocus => {
                 self.has_focus.set(true);
                 self.show_cursor(window);
+                window.show_virtual_keyboard(self.input_type());
             }
             FocusEvent::FocusOut | FocusEvent::WindowLostFocus => {
                 self.has_focus.set(false);
                 self.hide_cursor();
+                window.hide_virtual_keyboard();
```

Comment:
Another option be to combine this with `show_cursor()` and `hide_cursor()`. Just a thought, I think this is fine - just in case we add `show_cursor()` calls in the future then we also need to remember to call `show_virtual_keyboard()`.

Reply:
i thought about it, but there are other call to show_cursor for which one should not re-show the keyboard.